### PR TITLE
Add diagram for EC2 Auth flow

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

**Comment by joelthompson** [comment]
Sorry if this is overly nitpicky, but it's a bit misleading say that Vault talks to the "AWS Metadata Service." The instance talks to the EC2 instance metadata service that is accessible only to the EC2 instance, and Vault talks to the public EC2 API endpoints. I worry that the diagram as written will actually be somewhat confusing and misleading.

File: website/source/docs/auth/aws.html.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ for your use cases.
 
 ### EC2 Authentication Method
 
+[![Vault AWS Ec2 Authentication Flow](/assets/images/vault-aws-ec2-auth-flow.png)](/assets/images/vault-aws-ec2-auth-flow.png)
```
Comment (joelthompson) [inline website/source/docs/auth/aws.html.md]: s/Ec2/EC2/

Again, I realize I'm being a bit nitpicky, but I find it a bit distracting, and the rest of the document capitalizes the `C`.

**Comment by sethvargo** [comment]
> The instance talks to the EC2 instance metadata service that is accessible only to the EC2 instance, and Vault talks to the public EC2 API endpoints. I worry that the diagram as written will actually be somewhat confusing and misleading.

The challenge here is that the end user needs to understand that those two metadata services are connected at the cloud layer, at which point we start documenting AWS' architecture more than Vault's communication. I think we make this point clear in the text, and, while I always strive for technical accuracy, I don't know of a way to draw the diagram that doesn't add complexity. Where the metadata service lives is an implementation detail. One could easily argue that I should document subnets and vpcs in here, but my goal is to provide a simplistic overview of the data flow, particularly to non-technical or less-technical users who are trying to understand the workflow. In spending a lot of time with users and customers, the paragraphs of text don't "click" - they really benefit from having a visual aid.

**Comment by joelthompson** [comment]
My worry is that the diagram will be confusing to people who are very familiar with AWS networking but not so much Vault. With my security hat on, I would ask, "Does Vault need to talk to my instances' private metadata service? How will it do that? Or do all my instances need to talk to the public AWS EC2 APIs?"

I think a more accurate to representation of the abstractions AWS provides is to have the instance metadata service completely separate from the EC2 APIs, as two different nodes in the graph. Yes, they all connect under the hood in some way, but I think that's more of the implementation detail that can be omitted from the diagram.

**Comment by sethvargo** [comment]
> My worry is that the diagram will be confusing to people who are very familiar with AWS networking but not so much Vault.

My worry is the exact opposite. As written, the feedback I've gotten is that very few people understand how the data is flowing through the system. I think people who understand AWS will be able to read the paragraph and connect the dots. I think the answers to the questions you've posed are answered in the text below.

> I think a more accurate to representation of the abstractions AWS provides is to have the instance metadata service completely separate from the EC2 APIs, as two different nodes in the graph.

If you wanna sketch that up in omnigraffle or similar (or a napkin and take a picture), I'm happy to try and incorporate it. I can't think of a way to do that doesn't introduce additional confusion for users who aren't familiar with the AWS APIs.

**Comment by sethvargo** [comment]
I'm not sure what you were thinking in terms of a diagram, but I don't think this adds much value over the original diagram, especially given the narrative that follows.

**Comment by joelthompson** [comment]
Yeah, very similar to that. Thanks for sketching it up! (I'm really bad at graphics....) The main change is I would omit the dotted line between the EC2 metadata service and the public EC2 API service, as that's not really relevant to the data flow _at the time of authentication_.

Also, thanks for taking the time to engage with my feedback! We both share the same goal of making this easier for customers to understand, and I'm trying to offer my perspective of what I think your customers would find easier to consume. But I recognize that, as somebody who knows AWS EC2 very well and who knows the AWS authentication backend very well, I'm not exactly your target audience, and you of course know your customers better than I do. So if you think your original version will be more helpful for your customers, then I'll totally defer and you should go with it. I think either version would be great :)

**Comment by sethvargo** [comment]
Here's a version without the text on the image. I'm a bit conflicted, since having a lot of words on a photo is often hard to read. But not having the words makes the image on it's own less useful.

without text

**Comment by sethvargo** [comment]
Hey @joelthompson - I reworked this a bit. Please let me know what you think (and please verify that you approve of the new "listed" approach instead of paragraphs).

**Comment by joelthompson** [comment]
Hey @sethvargo -- sorry for not responding sooner, was travelling out of town. I think this looks great! One tiny tweak might be to add "Server" after "HashiCorp Vault" in the image, just to ensure people know it's not the Vault client/CLI tool (it's a bit ambiguous).

**Comment by sethvargo** [comment]
Updated

**Pull request merged**